dimod/quadratic/quadratic_model.py

```diff
@@ -462,17 +462,38 @@ def add_linear_from(self,
         """
         add_linear = self.data.add_linear
 
-        if default_vartype is not None:
-            default_vartype = as_vartype(default_vartype, extended=True)
-
         if isinstance(linear, Mapping):
             linear = linear.items()
 
-        for v, bias in linear:
-            add_linear(v, bias,
-                       default_vartype=default_vartype,
-                       default_lower_bound=default_lower_bound,
-                       default_upper_bound=default_upper_bound)
+        # checking whether the keyword arguments are present actually
+        # results in a pretty shocking performance difference, almost x2
```

Comment:
Are you sure the difference comes from _checking_ and not _propagating_?

Reply:
I meant checking improves the performance. I assume that it has something to do with the keyword arguments getting bound in the Cython code